src/engine/source/builder/builders/opBuilderHelperNetInfoAddress.cpp

```diff
@@ -0,0 +1,263 @@
+    BROADCAST
+};
+
+constexpr std::string_view getPath(Name field)
+{
+    switch (field)
+    {
+        case Name::ADDRESS: return "/address";
+        case Name::NETMASK: return "/netmask";
+        case Name::BROADCAST: return "/broadcast";
+        default: return "";
+    }
+}
+
+// when isIPv6 == true the third field will be set to 0, otherwise 1 for IPv4
+// agent <agent_id> netaddr save <scan_ID>|<name>|isIPv6=false|add[i]|netm[i]|broad[i]
+// agent 0001 netaddr save 1234|name|0|add0|netm0|broad0
+bool sysNetAddresTableFill(base::Event event,
+                           const std::string& agentId,
+                           const std::string& scan_id,
+                           const std::string& name,
+                           const std::string& ipObjectPath,
+                           std::shared_ptr<wazuhdb::WazuhDB> wdb,
+                           bool isIPv6)
+{
+    // Cheking if AddresArray exists
+    std::optional<std::vector<json::Json>> address_ar;
+    try
+    {
+        address_ar = event->getArray(ipObjectPath + getPath(Name::ADDRESS).data());
+    }
+    catch (const std::runtime_error& e)
+    {
+        return false;
+    }
+
+    if (!address_ar.has_value())
+    {
+        return false;
+    }
+
+    std::string middleFieldName = isIPv6 ? "IPv6" : "IPv4";
+
+    std::optional<std::vector<json::Json>> netmaskI;
+    try
+    {
+        netmaskI = event->getArray(ipObjectPath + getPath(Name::NETMASK).data())
+                       .value_or(std::vector<json::Json>());
+    }
+    catch (const std::runtime_error& e)
+    {
+        netmaskI = std::nullopt;
+    }
+
+    std::optional<std::vector<json::Json>> broadcastI;
+    try
+    {
+        broadcastI = event->getArray(ipObjectPath + getPath(Name::BROADCAST).data())
+                         .value_or(std::vector<json::Json>());
+    }
+    catch (const std::runtime_error& e)
+    {
+        broadcastI = std::nullopt;
+    }
+
+    for (size_t i = 0; i != address_ar.value().size(); ++i)
+    {
+        std::optional<std::string> addressIValue {};
+        std::optional<std::string> netmaskIValue {"NULL"};
+        std::optional<std::string> broadcastIValue {"NULL"};
+
+        try
+        {
+            addressIValue = address_ar.value().at(i).getString();
+        }
+        catch (const std::out_of_range& e)
+        {
+            return false;
+        }
+
+        if (netmaskI.has_value())
+        {
+            try
+            {
+                netmaskIValue = netmaskI.value().at(i).getString();
+            }
+            catch (const std::out_of_range& e)
+            {
+                netmaskIValue = "NULL";
+            }
+        }
+
+        if (broadcastI.has_value())
+        {
+            try
+            {
+                broadcastIValue = broadcastI.value().at(i).getString().value_or("NULL");
+            }
+            catch (const std::out_of_range& e)
+            {
+                broadcastIValue = "NULL";
+            }
```

Comment:
Is it possible to unify and use the value_or in the query? or outside the query?

Reply:
Im not finding a way to unify this process without using a function or a lambda with only two possible arguments. Resulting in lesser readabillity of the code and none performance